src/Hedgehog/Extras/Test/Golden.hs

```diff
@@ -43,16 +45,66 @@ mGoldenFileLogFile :: Maybe FilePath
 mGoldenFileLogFile = IO.unsafePerformIO $
   IO.lookupEnv "GOLDEN_FILE_LOG_FILE"
 
--- | Whether the test should create the golden files if the file does ont exist.
-createFiles :: Bool
-createFiles = IO.unsafePerformIO $ do
+-- | Whether the test should create the golden files if the files do not exist.
+createGoldenFiles :: Bool
+createGoldenFiles = IO.unsafePerformIO $ do
```

Comment:
Do you need the `unsafePerformIO`? Aren't all callers in `MonadIO m` anyway?

Reply:
It's mainly for memoisation.

Reply:
I'm tempted to say it's premature optimization, but you know the Haskell etiquette way better than I do, so your call!